solr/core/src/java/org/apache/solr/packagemanager/SolrPackage.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.packagemanager;
+
+
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.solr.common.annotation.JsonProperty;
+import org.apache.solr.common.util.ReflectMapWriter;
+
+/**
+ * Describes a package (along with all released versions) as it appears in a repository.
+ */
+public class SolrPackage implements Comparable<SolrPackage>, ReflectMapWriter {
+
+  @JsonProperty("name")
+  public String name;
+
+  @JsonProperty("description")
+  public String description;
+
+  @JsonProperty("versions")
+  public List<SolrPackageRelease> versions;
+
+  @JsonProperty("repository")
+  private String repository;
+
+  @Override
+  public String toString() {
+    return jsonStr();
+  }
+
+  public static class SolrPackageRelease implements ReflectMapWriter {
+    @JsonProperty("version")
+    public String version;
+
+    @JsonProperty("date")
+    public Date date;
+
+    @JsonProperty("artifacts")
+    public List<Artifact> artifacts;
+
+    @JsonProperty("manifest")
```

Comment:
Won't this class represent the content of `manifest.json`? In that case it could be misleading to have a property `manifest` inside? Perhaps I misunderstand?

Reply:
This represents a version / release. The manifest is inside (either explicitly specified in the repository.json or contained within the jar). Here's an example: https://github.com/apache/lucene-solr/pull/994/files#diff-ab1e50fd5605ee1a39764390b9d4faeeR6-R42